src/pybind/mgr/nfs/export.py

```diff
@@ -666,8 +682,10 @@ def create_export_from_dict(self,
                 raise FSNotFound(fs_name)
 
             validate_cephfs_path(self.mgr, fs_name, path)
+            if fsal["cmount_path"] != "/":
+                _validate_cmount_path(self.mgr, fsal["cmount_path"], path)  # type: ignore
 
-            user_id = f"nfs.{cluster_id}.{ex_id}"
+            user_id = f"nfs.{generate_unique_id(cluster_id, fs_name, fsal['cmount_path'])}"
```

Comment:
the working mechanism of `generate_unique_id()` is on point but the name `generate_unique_id` would make code readers think it is creating a user with a "unique" id on every call. This can either be fixed by adding a comment or maybe changing the name to something self-explanatory.

Reply:
I can do generate_user_id if that makes sense?

Reply:
there can be something as simple as `get_user_id` but i'm on fence here.

Reply:
Sounds good to me, will change it!